homeassistant/components/octoprint.py

```diff
@@ -11,41 +11,50 @@
 import voluptuous as vol
 from aiohttp.hdrs import CONTENT_TYPE
 
-from homeassistant.const import CONF_API_KEY, CONF_HOST, CONTENT_TYPE_JSON
+from homeassistant.const import (CONF_API_KEY, CONF_HOST, CONTENT_TYPE_JSON,
+                                 CONF_NAME, CONF_PORT)
 import homeassistant.helpers.config_validation as cv
 
 _LOGGER = logging.getLogger(__name__)
 
 DOMAIN = 'octoprint'
 CONF_NUMBER_OF_TOOLS = 'number_of_tools'
 CONF_BED = 'bed'
+DEFAULT_NAME = 'OctoPrint'
 
 CONFIG_SCHEMA = vol.Schema({
-    DOMAIN: vol.Schema({
+    DOMAIN: vol.All(cv.ensure_list, [vol.Schema({
         vol.Required(CONF_API_KEY): cv.string,
         vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_PORT, default=80): cv.positive_int,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
```

Comment:
Maybe it would be more natural to make the config a dictionary with the names as keys and value be another dictionary with the rest of the config options. Then the names would naturally be unique.

Or skip the requirement on unique name and use base url as storage key instead.

Reply:
I would like to keep the name optional, for two reasons:
 * ease of migrating from the current version
 * the vast majority of users will have only one printer and would probably be fine with having it named "Octoprint"